### PR TITLE
Move `SenExprBuilder` to a header file

### PR DESCRIPTION
**Pull request by kbieganski**
Moves `SenExprBuilder` to a common header file. No functional change intended. 

Pre-PR to #3599.

Signed-off-by: Krzysztof Bieganski <kbieganski@antmicro.com>

### CODE REVIEW

File: src/V3SenTree.h [hunk 1]
```diff
@@ -97,4 +97,221 @@ class SenTreeFinder final {
     }
 };
 
+//######################################################################
+// SenExprBuilder constructs the expressions used to compute if an
+// AstSenTree have triggered
+
+class SenExprBuilder final {
```
Comment (gezalore) [inline src/V3SenTree.h:104]: Let's put this into `V3SenExprBuilder.h`

Comment (kbieganski) [reply src/V3SenTree.h]: Done.

File: src/V3Timing.cpp [hunk 2]
```diff
@@ -50,6 +50,7 @@
 #include "V3Const.h"
 #include "V3EmitV.h"
 #include "V3Graph.h"
+#include "V3SenExprBuilder.h"
```
Comment (gezalore) [inline src/V3Timing.cpp:53]: This should be in a later patch I believe.

Comment (kbieganski) [reply src/V3Timing.cpp]: Correct, fixed.

**Pull request merged**